### DISCUSSION TITLE
Where is the Go language v1.22.0 update?

### DISCUSSION THREAD

**Discussion by setanarut** [Homebrew/discussions#5117] [Tap maintenance and Homebrew development]
brew install go

**Comment by SMillerDev** [comment]
On the go homepage, or if you want it through brew it will be available when all the volunteers have put in enough of their free time to get it to work.

**Reply by setanarut** [reply]
Why is this done manually? Shouldn't it be automatic?

**Reply by alebcay** [reply]
You can follow progress at https://github.com/Homebrew/homebrew-core/pull/157782.

> Why is this done manually? Shouldn't it be automatic?

If the only purpose of Homebrew was to distribute `go`, then absolutely - we could just make a script that, upon seeing a new Go version be released, update the installation process to distribute the latest version of `go`.

But Homebrew does not only distribute `go`. Homebrew also distributes a bunch of packages that build from `go`.

To your point, much of the update process is automatic. There is automation to build the new Go version and to test that it works on all of our supported platforms (several macOS versions, on both Intel and Apple Silicon, and Linux on Intel). It also tests that all of the dependent packages are also capable of building using the new version of Go (and that they work properly, at least for an initial smoke test).

Even assuming that everything goes smoothly, this can still take hours to a day of CI time for one pass. Then consider that some packages may not build using the new version of Go. Triage and investigation needs to occur, and deciding what to do about it - do we keep the broken packages on the old version of Go or, if a fix from upstream is imminent, do we just wait for it instead?

Then once some/many of the issues are addressed, we run another CI pass and it may take another while to confirm that the dependents' issues are fixed.

You can see some of the outstanding hold-ups and issues in the pull request I linked above.

---

To summarize - the remaining manual parts are mostly investigating what isn't working and deciding how to proceed with fixing it. All of the verification and build steps are automated (and even when automated, takes a long time).

Sure, we could also just say 'screw the dependents, just ship it', but 'go fast and break things' tends to come back to bite us later.

If you _absolutely_, _positively_, ***must*** have the shiniest new version of Go 1.22 (that is, you really need [these features](https://tip.golang.org/doc/go1.22)) and you cannot wait, you could also edit the `go` formula on your local machine (`brew edit go`) and then install it yourself from the local copy (`HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source go`) - note that you may have to blow away these changes later if you have trouble updating, with `HOMEBREW_NO_INSTALL_FROM_API=1 brew update-reset`.

You (or someone else) could also host a standalone `go` formula in another [tap](https://docs.brew.sh/Taps) which receives the latest version as soon as possible. With no other software in Homebrew depending on it, it could be updated as fast as you wish.

Or you may as well install it from the official upstream distribution, available at https://go.dev/dl/.

**Reply by setanarut** [reply]
I had already uninstalled it and updated it with the installer on the website. The primary task of a package manager is to make installation easy. This does not include cheat updating.

**Reply by SMillerDev** [reply]
> The primary task of a package manager is to make installation easy.

No, it is to manage packages. That includes the packages you might not be interested in.
